### DISCUSSION TITLE
[插件分享] 写了个插件解决 module tool 无法转换动态导入类型的别名

### DISCUSSION THREAD

**Discussion by baranwang** [web-infra-dev/modern.js#4511] [Show and tell]
[别名转换](https://modernjs.dev/module-tools/guide/advance/in-depth-about-build.html#%E5%88%AB%E5%90%8D%E8%BD%AC%E6%8D%A2) 中提到无法转换动态引入类型的别名

> 对于类似 Promise<import('@common/utils')> 这样形式的输出类型目前无法进行转换。

但是 `npm-component` 构建模式会经常生成带别名的类型，为了避免这个问题，整个项目都要用相对路径来书写，会让代码可读性变差。

于是我实现了一个基于 [`typescript-transform-paths`](https://www.npmjs.com/package/typescript-transform-paths) 插件的解决方案，通过构建时向 tsconfig 添加插件，构建结束再还原 tsconfig 的方式解决了这个问题。

[modern-js-plugin-module-transform-paths](https://www.npmjs.com/package/modern-js-plugin-module-transform-paths) 

同时也可以作为一个思路给 webinfra 团队的同学，看看不是有可能有入侵性更小的方案。

**Comment by chenjiahan** [comment]
感谢分享 ❤️

**Comment by 10Derozan** [comment]
I will restart the evaluation of this feature and try to build it into the feature,  and then remove babel.
